metrics-api/src/main/java/org/killbill/commons/metrics/api/Counter.java

```diff
@@ -20,21 +20,21 @@
 /**
  * An incrementing and decrementing counter metric.
  */
-public interface Counter extends Metric, Counting {
+public interface Counter extends Metric {
 
     /**
      * Increment the counter by {@code n}.
      *
      * @param n the amount by which the counter will be increased
      */
-    void inc(long n);
+    void inc(double n, String ...labels);
```

Comment:
How about `BigDecimal` instead of `double`? That was we always ensure precision?

Reply:
I'm not sure we care enough about precision or rounding for metrics to justify the added interaction complexity `BigDecimal` introduces.

Reply:
I think we should still go with `BigDecimal`. This would add more flexibility (e.g. allow decimal values).

Reply:
No underlying implementation would support BigDecimal

Reply:
Talked to @sbrossie -- 👍 for `double`.